plugins/woocommerce/includes/admin/class-wc-admin-reports.php

```
@@ -31,6 +31,31 @@ public static function register_hook_handlers() {
+		if ( $defer ) {
+			// Schedule the deletion, cap the execution to single pending event at any given time.
+			$scheduled = as_has_scheduled_action( 'woocommerce_delete_legacy_report_transients' );
+			if ( ! $scheduled ) {
+				as_schedule_single_action( time() + MINUTE_IN_SECONDS, 'woocommerce_delete_legacy_report_transients', array( $order_id, false ), 'woocommerce' );
+			}
+			return;
+		}
+
+		delete_transient( 'wc_admin_report' );
+		foreach ( self::get_reports() as $report_group ) {
+			foreach ( $report_group['reports'] as $report_key => $report ) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$report'. (undefined)

(UnusedLocalVariable)